### PR TITLE
Arm32 support

### PR DESCRIPTION
**Pull request by etienne-lms**
Supersedes https://github.com/ARM-software/arm-systemready/pull/25.

### CODE REVIEW

**Comment by etienne-lms** [comment]
This change is related to https://github.com/ARM-software/bbr-acs/pull/11. I've introduced a hacky commit to get this series building with that change. See commit "[HACK]get bbr-acs.git from etienne-lms repo".

This P-R currently fails to build FWTS for I reason I haven't investigated yet, hence a hacky commit to disable FWTS images build, see commit "[HACK] don't build fwts: builld issue to fix".

File: common/scripts/build-all-arm.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage $0 <ES|IR> F"
```
Comment (vstehle) [inline common/scripts/build-all-arm.sh:32]: You can drop ES as only IR supports 32b.

File: common/scripts/build-fwts-arm.sh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    echo $BBR_DIR
+
+    if [ "$BUILD_PLAT" = "IR" ]; then
+        if ! patch -R -s -f --dry-run -p1 < $BBR_DIR/ebbr/config/IR_VER.patch ; then
+            echo "Applying FWTS Patch ..."
+            patch -p1 < $BBR_DIR/ebbr/config/IR_VER.patch
+        fi
+    fi
+    if [ "$BUILD_PLAT" = "ES" ]; then
```
Comment (vstehle) [inline common/scripts/build-fwts-arm.sh:90]: You can drop ES as only IR supports 32b.

File: common/scripts/make_image-arm.sh [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+exit_fun() {
+   exit 1 # Exit script
+}
+
+BUILD_PLAT=$1
+
+if [ -z "$BUILD_PLAT" ]
+then
+   echo "Specify platform ES, IR or SR"
```
Comment (vstehle) [inline common/scripts/make_image-arm.sh:202]: You can drop ES and SR as only IR supports 32b.

File: common/scripts/build-all-arm.sh (hunk 1 above)
Comment (etienne-lms) [reply common/scripts/build-all-arm.sh:32]: OK.
Given that we mandate target IR and type F, do we really want to pass those arguments (to keep synced with the 64b scripts), or can we simply drop these 2 args?

**Review by etienne-lms — COMMENTED** [review]
Updated with fixup commits to clean up.
There is still things to clean (really needs PLAT/TYPE args?)
Synced with https://github.com/ARM-software/bbr-acs/pull/11.

File: common/scripts/build-all-arm.sh (hunk 1 above)
Comment (edhay) [reply common/scripts/build-all-arm.sh]: I suggest, we can drop both the args.

File: common/scripts/get_source.sh [hunk 4]
```diff
@@ -175,7 +187,9 @@ get_bbr_acs_src()
    if [ -z $ARM_BBR_TAG ]; then
        #No TAG is provided. Download the latest code
        echo "Downloading Arm BBR source code."
-       git clone  --depth 1 https://github.com/ARM-software/bbr-acs.git bbr-acs
+       echo "                BBR source code from etienne-lms (P-R in bbr-acs/git)"
```
Comment (edhay) [inline common/scripts/get_source.sh:190]: Todo: This code to be changed after BBR PR is accepted.

File: common/scripts/get_source.sh [hunk 5]
```diff
@@ -189,10 +203,12 @@ sudo apt install git curl mtools gdisk gcc\
 
 get_uefi_src
 
-if [ $BAND == "SR" ]; then
-    get_sbsa_src
-else
-    get_bsa_src
+if [ $TARGET_ARCH == "arm" ]; then
```
Comment (edhay) [inline common/scripts/get_source.sh:206]: This check may not be needed

File: common/scripts/get_source.sh [hunk 6]
```diff
@@ -30,6 +30,11 @@
 
 TOP_DIR=`pwd`
 
+#Optional argument 'arm' shall be set when targeting a 32bit Arm device
```
Comment (edhay) [inline common/scripts/get_source.sh:33]: Initialise 
TARGET_ARCH="aarch64"

else the script will give error in line 175 check as $TARGET_ARCH is NULL.

  if [ $TARGET_ARCH != "arm" ]; then

File: common/scripts/get_source.sh (hunk 5 above)
Comment (etienne-lms) [reply common/scripts/get_source.sh:206]: done

File: common/scripts/get_source.sh (hunk 6 above)
Comment (etienne-lms) [reply common/scripts/get_source.sh:33]: done

File: IR/scripts/build-scripts/build-ir-live-image-arm.sh [hunk 7]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply IR/scripts/build-scripts/build-ir-live-image-arm.sh:1]: Do you mean an inline description in this script file or changes in _IR/README.md_?

**Review by etienne-lms — COMMENTED** [review]
IR/README.md

File: IR/scripts/build-scripts/build-ir-live-image-arm.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (edhay) [reply IR/scripts/build-scripts/build-ir-live-image-arm.sh:1]: We could add a new file "Build Instructions for 32bit IR ACS.md"

File: common/scripts/get_source.sh [hunk 8]
```diff
@@ -194,6 +210,7 @@ if [ $BAND == "SR" ]; then
 else
     get_bsa_src
 fi
+
```
Comment (edhay) [inline common/scripts/get_source.sh:213]: On line 205, please add the below code:
if [ $TARGET_ARCH == "arm" ]; then
    sudo apt-get install libmpc-dev
fi

Resolution of Linux build issue seen in some envs

Comment (etienne-lms) [reply common/scripts/get_source.sh:213]: +1. I'll append a fixup patch to the series. I'll squash the fixup commit at next rebase of this P-R.

**Pull request merged**